mariadb/templates/mariadb-deployment.yaml

```diff
@@ -1,40 +1,36 @@
-apiVersion: v1
-kind: ReplicationController
+apiVersion: extensions/v1beta1
+kind: Deployment
 metadata:
-  name: {{ printf "%s-%s" .Release.Name "mariadb" | trunc 24 }}
+  name: {{ template "fullname" . }}
   labels:
```

Comment:
I would remove `provider` and use `chart`. Open to keeping `version`, but might be better to just remove it.